client/termsetting/handlers/geneExpression.ts

```diff
@@ -25,16 +26,24 @@ export async function getHandler(self) {
 }
 
 export async function fillTW(tw: GeneExpressionTW, vocabApi: VocabApi, defaultQ: NumericQ | null = null) {
-	if (!tw.q?.mode) tw.q = { mode: 'continuous' }
-	const mode = tw.q.mode || 'continuous'
+	if (typeof tw.term.gene != 'string' || !tw.term.gene) throw 'geneExpression tw.term.gene must be non-empty string'
+	if (!tw.term.name) tw.term.name = tw.term.gene // auto fill if .name is missing
+
+	if (!tw.q?.mode) tw.q = { mode: 'continuous' } // supply default q if missing
+	if (defaultQ) copyMerge(tw.q, defaultQ) // override if default is given
 
 	if (!tw.term.bins) {
+		/* gene term is missing bin definition, this is expected as it's not valid to apply same bin to genes with vastly different exp range, and not worth it to precompute each gene's default bin with its actual exp data
+		here make a request to determine default bin for this term based on its data
+		(in gdc this adds significant pause when adding gene exp term to oncomatrix)
+		*/
 		const defaultBins = await vocabApi.getDefaultBins({ tw })
 		if ('error' in defaultBins) throw defaultBins.error
 		tw.term.bins = defaultBins
-		tw.q = JSON.parse(JSON.stringify(tw.term.bins.default))
+		const currMode = tw.q.mode // record current mode before q{} is overriden
+		tw.q = structuredClone(tw.term.bins.default)
+		tw.q.mode = currMode
 	}
-	tw.q.mode = mode
```

Comment:
the q mode needs to be set not just when initializing the bins. When a discrete mode is specified also. This can be tested for example opening the barchart with age and trying to overlay geneExpression in termdbTest. Now it goes directly to the violin plot, instead of doing overlay of the geneExpression in discrete mode.